libr/asm/arch/8051/8051_ass.c

```diff
@@ -542,6 +542,20 @@ static int register_number(char const*reg) {
 	return 8; // not register 0-7, so...
 }
 
+/**
+ * translate a register name to an offset for an instruction
+ * failure case is 10
+ */
+static ut8 register_offset(const char *reg) {
+	if (!r_str_casecmp (reg, "@r0") || !r_str_casecmp (reg, "[r0]")) {
+		return 0;
+	} else if (!r_str_casecmp (reg, "@r1") || !r_str_casecmp (reg, "[r1]")) {
```

Comment:
don't else after return
```suggestion
	}
	if (!r_str_casecmp (reg, "@r1") || !r_str_casecmp (reg, "[r1]")) {
```